[TACHYON-540] Worker File Level UI

carsonwang opened the pull request:
https://tachyon.atlassian.net/browse/TACHYON-540
Provide a worker level web page that displays the worker's cached blocks at a file level.

calvinjia commented on servers/src/main/java/tachyon/web/WebInterfaceWorkerBlockInfoServlet.java:
```diff
@@ -201,19 +201,17 @@ private UiFileInfo getUiFileInfo(TachyonFS tachyonClient, int fileId, TachyonURI
+        // The block last access time is not available. Use -1 for now.
+        uiFileInfo.addBlock(storageLevelAlias, blockId, blockSize, -1);
+      }
+    }
```
I think this can be simplified by just directly trying `getBlockMeta`. I don't think `FileDoesNotExistException` is required to be thrown and can just be the `IOException` thrown by `getBlockMeta` for now.

carsonwang replied:
@calvinjia , the method only displays the blocks of the file that exist on the worker. It is possible that some of the blocks do not exist. In this case, I think we don't want to throw the IOException. An exception will only be thrown when there is no blocks at all.

calvinjia replied:
Oh okay, that makes sense.

calvinjia commented on servers/src/main/java/tachyon/worker/block/BlockDataManager.java:
```diff
@@ -430,6 +430,27 @@ public void userHeartbeat(long userId, List<Long> metrics) {
+   * @return metadata of the block or null
+   * @throws IOException if no BlockMeta for this blockId is found
+   */
+  public BlockMeta getBlockMeta(long blockId) throws IOException {
```
Could you rename this to something like `getVolatileBlockMeta`? @apc999 do you have any preferences for the naming convention?

calvinjia commented on servers/src/main/java/tachyon/worker/block/BlockDataManager.java:
```diff
@@ -430,6 +430,27 @@ public void userHeartbeat(long userId, List<Long> metrics) {
+   */
+  public boolean hasBlockMeta(long blockId) {
+    return mBlockStore.hasBlockMeta(blockId);
+  }
```
I think we can remove this method, see my comment in the servlet code.

calvinjia commented:
@carsonwang thanks for the PR, I've left a few comments.

calvinjia commented on servers/src/main/java/tachyon/worker/block/BlockDataManager.java:
```diff
@@ -430,6 +430,27 @@ public void userHeartbeat(long userId, List<Long> metrics) {
   }
 
   /**
+   * Gets the metadata of a block given its blockId or throws IOException.
```
Could you document this in more detail (ie. how its different from `getBlockMeta`)?

calvinjia commented:
Looks good overall, just a minor comment about the javadoc.

carsonwang commented:
Thanks @calvinjia  for reviewing this. I updated the javadoc and also renamed the method name in BlockStore and BlockDataManager. Let me know if this looks fine.

calvinjia commented:
+1 LGTM, please resolve the merge conflicts.

haoyuan commented:
We can’t automatically merge this pull request.

carsonwang commented:
@haoyuan ,merge conflict is resolved.

apc999 commented:
+1 LGTM. Thanks for the patch!

haoyuan commented:
Thanks @carsonwang ; Could you please post a snapshot here?

carsonwang commented:
@haoyuan , screenshot is attached. The only change to the UI is that the last access time of the block is removed.
![tachyonworker](https://cloud.githubusercontent.com/assets/9278199/8611161/490b04fa-26ef-11e5-8c00-5e46d668673c.png)

calvinjia commented:
@carsonwang thanks for posting the screenshot, could you resolve the merge conflicts please?
